### PR TITLE
TF2 porting: tagger decoder cleanup

### PR DESCRIPTION
**Pull request by jimthompson5802**
# Code Pull Requests

This should finish off the Tagger decoder
* todo for feedforward attention
* add time series support
* add regularizer specification

### CODE REVIEW

File: ludwig/models/modules/sequence_decoders.py [hunk 1]
```diff
@@ -149,104 +149,56 @@ class SequenceTaggerDecoder(Layer):
     def __init__(
             self,
             initializer=None,
```
Comment (w4nderlust) [inline ludwig/models/modules/sequence_decoders.py:151]: this doesn't seem to be used, can remove

File: ludwig/models/modules/sequence_decoders.py [hunk 2]
```diff
@@ -149,104 +149,56 @@ class SequenceTaggerDecoder(Layer):
...
-            regularize=True,
+            use_bias=True,
+            kernel_initializer='glorot_uniform',
```
Comment (w4nderlust) [inline ludwig/models/modules/sequence_decoders.py:153]: although Dense uses ths name, throughout the rest of the code we are always calling this weights_initializer. The same  applies to kernel_regularizer too.

File: ludwig/models/modules/sequence_decoders.py [hunk 3]
```diff
@@ -149,104 +149,56 @@ class SequenceTaggerDecoder(Layer):
...
+            bias_initializer='zeros',
+            kernel_regularizer=None,
+            bias_regularizer=None,
+            activity_regularizer=None,
             attention=False,
             num_classes=0,
```
Comment (w4nderlust) [inline ludwig/models/modules/sequence_decoders.py:159]: this should probably be the first parameter and it should not be optional, as 0 num classes does not make sense.

File: ludwig/models/modules/sequence_decoders.py [hunk 4]
```diff
@@ -149,104 +149,56 @@ class SequenceTaggerDecoder(Layer):
...
-            loss=None,
+            is_timeseries=None,
```
Comment (w4nderlust) [inline ludwig/models/modules/sequence_decoders.py:160]: this should be a bool, so default is False

File: ludwig/models/modules/sequence_decoders.py [hunk 5]
```diff
@@ -149,104 +149,56 @@ class SequenceTaggerDecoder(Layer):
...
     ):
         super(SequenceTaggerDecoder, self).__init__()
         self.initializer = initializer
-        self.regularize = regularize
         self.attention = attention
 
-        # todo tf2 add other required parameters for Dense layer
-        self.decoder_layer = Dense(num_classes)
+        if is_timeseries:
+            units = 1
```
Comment (w4nderlust) [inline ludwig/models/modules/sequence_decoders.py:168]: would simplify to `num_classes = 1` and remove the else

File: ludwig/models/modules/sequence_decoders.py [hunk 6]
```diff
@@ -149,104 +149,56 @@ class SequenceTaggerDecoder(Layer):
...
+        else:
+            units = num_classes
+
+        self.decoder_layer = Dense(
+            units,
```
Comment (w4nderlust) [inline ludwig/models/modules/sequence_decoders.py:173]: would change to `units=num_classes`

File: ludwig/models/modules/sequence_decoders.py [hunk 7]
```diff
@@ -149,104 +149,56 @@ class SequenceTaggerDecoder(Layer):
...
-            # hidden_size,
-            # regularizer,
-            # is_timeseries=False
+            inputs,
+            training=None,
+            mask=None
     ):
-        logger.debug('  hidden shape: {0}'.format(hidden.shape))
-        if len(hidden.shape) != 3:
+        logger.debug('  hidden shape: {0}'.format(inputs.shape))
```
Comment (w4nderlust) [inline ludwig/models/modules/sequence_decoders.py:188]: I'm removing all this loggig calls from call() everywhere for now

File: ludwig/models/modules/sequence_decoders.py (hunk 1 above)
Comment (jimthompson5802) [reply ludwig/models/modules/sequence_decoders.py:151]: done

File: ludwig/models/modules/sequence_decoders.py (hunk 2 above)
Comment (jimthompson5802) [reply ludwig/models/modules/sequence_decoders.py:153]: done

File: ludwig/models/modules/sequence_decoders.py (hunk 4 above)
Comment (jimthompson5802) [reply ludwig/models/modules/sequence_decoders.py:160]: done

File: ludwig/models/modules/sequence_decoders.py (hunk 5 above)
Comment (jimthompson5802) [reply ludwig/models/modules/sequence_decoders.py:168]: done

File: ludwig/models/modules/sequence_decoders.py (hunk 7 above)
Comment (jimthompson5802) [reply ludwig/models/modules/sequence_decoders.py:188]: done

File: ludwig/models/modules/sequence_decoders.py (hunk 6 above)
Comment (jimthompson5802) [reply ludwig/models/modules/sequence_decoders.py:173]: done

File: ludwig/models/modules/sequence_decoders.py (hunk 3 above)
Comment (jimthompson5802) [reply ludwig/models/modules/sequence_decoders.py:159]: done

**Pull request merged**